### PR TITLE
Feature: Add custom auto-link header

### PR DESCRIPTION
**Pull request by bdenham**
## Description

This PR allows writers to customize the anchor links for headings h2 – h5 using `{#my-custom-link}` syntax.

## Related Issue

COMDOX-272

## Motivation and Context

Commerce GraphQL documentation has loads of custom anchor links for headers. This addition should provide a seamless migration for those and other docs that use custom header links.

## How Has This Been Tested?

I tested using the example project within the theme and against the commerce-frontend-core repo. 

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -310,6 +316,19 @@ You can exclude pages from the build by either moving them out of `src/pages` or
...
+By default, the theme automatically generates an anchor link for headings h2 – h6, using the kabab-case (lowercase words separated by hyphens).
+For example, if the heading name is **Inline code example**, the theme creates the following anchor link: `#inline-code-example`.
+
+You can customize the anchor link for a heading using the following syntax:
+
+```markdown
+## Inline code example {#my-custom-anchor-link}
+```
+
+This will override the auto-link behavior and create a custom anchor link.
```
Comment (keharper) [inline README.md:330]: Mention or illustrate that you don't have to maintain kabab-case in the anchor link.

File: example/src/pages/guides/index.md [hunk 2]
```diff
@@ -12,14 +12,14 @@ Get Started
...
-Authentication
+Authentication {#my-authentication}
 ---
 
 There are multiple type of authentication for authenticating with the analytics APIs. General information
 about authenticating with Adobe's APIs is located here. Specific information related to authenticating
 with the analytics APIs is available in the sections below
 
-## OAuth
+## OAuth {#OAuthentication}
```
Comment (keharper) [inline example/src/pages/guides/index.md:22]: I don't understand why you're changing anchors here. OAuth is a common term. OAuthentication seems to refer to a very specific package. Also, I don't see any links going to this heading.

Comment (bdenham) [reply example/src/pages/guides/index.md:22]: @keharper this is just testing material. not intended to mean anything other than me adding custom links to various places in the meaningless example test docs. :)

File: README.md (hunk 1 above)
Comment (bdenham) [reply README.md]: Good call. Done.

**Review by keharper — APPROVED** [review]

File: example/package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "name": "example",
   "version": "0.0.1",
   "dependencies": {
-    "@adobe/gatsby-theme-aio": "*",
+    "@adobe/gatsby-theme-aio": "4.3.0-rc6",
```
Comment (dmitrymatio) [inline example/package.json:6]: Can be reverted to`*` ?

Comment (bdenham) [reply example/package.json:6]: @dmitrymatio, for some reason, `*` was not working earlier. I kept making code changes and doing another build to see no results until I changed it to match our current version. I'll try again now.

Comment (bdenham) [reply example/package.json:6]: Oh, duh...yeah I'll revert. Covid brain...I had to temporarily do that during dev. Doh!

**Review by dmitrymatio — APPROVED** [review]

**Pull request merged**